src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Linux.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Net;
+using System.Collections;
+using System.ComponentModel;
+using System.Text;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Threading;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.DirectoryServices.Protocols
+{
+    public partial class LdapConnection
+    {
+        private void InternalInitConnectionHandle(string hostname)
+        {
+            OpenLDAP.ldap_initialize(out IntPtr ldapServerHandle, $"ldap://{hostname}:{((LdapDirectoryIdentifier)_directoryIdentifier).PortNumber}/");
```

Comment:
>ldap_initialize [](start = 21, length = 15)

we don't need to check if this is failed?

Reply:
We do validate it on the constructor for ConnectionHandle which is called right after this when we pass in the handle. This is the way it works in Windows as well.